llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -1948,11 +1948,20 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
     break;
 
   case Intrinsic::cttz:
-  case Intrinsic::ctlz:
+  case Intrinsic::ctlz: {
+    // If ctlz/cttz is only used as a shift amount, set is_zero_poison to true.
+    bool Changed = false;
+    if (II->hasOneUse() && match(II->getArgOperand(1), m_Zero()) &&
+        match(II->user_back(), m_Shift(m_Value(), m_Specific(II)))) {
+      replaceOperand(*II, 1, Builder.getTrue());
+      Changed = true;
+    }
```

Comment:
Any particular reason for this code to be outside foldCttzCtlz?

I don't think this micro-optimization of changing the flag and running the following opts in one pass is really needed.

Reply:
Done.